### DISCUSSION TITLE
Linearization: effect of disabling pitch and generator&torque control on system dynamics

### DISCUSSION THREAD

**Discussion by GertHeirman** [OpenFAST/openfast#2545] [Q&A]
I am trying to add an additional controller to a wind turbine (to be more precise: an active tower damper to reduce side-side tower oscillations).
I would like to add this to a realistic turbine model: I have chosen the 10 MW reference turbine (https://github.com/IEAWindTask37/IEA-10.0-198-RWT). The additional controller should operate in parallel to the existing controller logic.

I want to extract a linearized model to design/choose reasonable parameters for this new controller: e.g. the gains of a PID.
To linearize, I seem to be forced to switch off the controllers that are part of the 10 MW reference wind turbine.
As described in other forum topics:
PCMode    5 --> 0
VSContrl    5 --> 1    +    a reasonable number for VS_RtTq

I switch off the original controllers with quite some reluctance, since I expect that the existing controllers will have an effect on the overall system dynamics.
So I fear to end up with a linearized model with different dynamics as the system for which I am designing an additional controller.
Furthermore, when looking at the time-domain simulation results, the _original model_ and _the one modified to allow for linearization_ converge to different operating points (for the same constant wind speed etc. I end up with a.o. a different rotor speed after a long time (200 s) ). A bit worrying...

Hence my questions: 
What is exactly the effect of switching off the external pitch control and generator and torque control on the system dynamics?
To which extent will the dynamics be modified by those different settings?
Can I somehow linearize the system model with its original controllers?

**Comment by jjonkman** [comment]
Dear @GertHeirman,

The control certainly has an impact on the system response, so, neglecting the controller in the linearization will certainly have an impact.  The challenge with the Bladed-style DISCON controller interface (`PCMode` = `VSContrl` = 5) is that this interface does not follow the OpenFAST modularization framework, so, while OpenFAST can see the inputs and outputs to this controller, it cannot see what is inside the controller, and thus, OpenFAST is unable to linearize it.  OpenFAST can be linearized when enabling the built-in control options (like `VSContrl` = 1).  You can also augment the linearized model derived from OpenFAST with control effects such as PI pitch control in a post-processing step.

The fact that your linearized and time-domain models result in different rotor speed for the same wind speed suggests to me that you did not arrive at the same generator torque and blade-pitch angles in the linearized solution that you did in the time-domain model.  As you said, you can set `VS_RtTq` yourself with `VSContrl` = 1 in ServoDyn (and presumably you've set `VS_RtGnSp` = `VS_Rgn2K` = `VS_SlPc` to small values, such as 9.999E-9), which is normally used in conjunction with `CalcSteady` = TRUE and `TrimCase` = 3 in the OpenFAST primary _.fst_ input file.  Is that what you've set and is your `VS_RtTq` set to be the same generator torque seen in the time-domain model?  In that case, I would expect `TrimCase` = 3 to arrive at the same blade pitch as in your time-domain model.

Best regards,
